src/api/java/com/minecolonies/api/colony/buildings/modules/AbstractBuildingModule.java

```diff
@@ -0,0 +1,139 @@
+package com.minecolonies.api.colony.buildings.modules;
+
+import com.minecolonies.api.colony.ICitizenData;
+import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.buildings.IBuilding;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Abstract class for all modules. Has base methods for all the necessary methods that have to be called from the building.
+ */
+public abstract class AbstractBuildingModule implements IBuildingModule
```

Comment:
This class is weird AF.
This is just an empty class which literally makes no sense.

If you want this to be the default behaviour for those empty methods for a module, then make it the default implementation of the method in the interface not in this weird abstract class.

Reply:
Yeah, true, I forgot I can set the empty methods as defaults in the interface.